### DISCUSSION TITLE
Opening modal changes my underlying page container width

### DISCUSSION THREAD

**Discussion by dprothero** [twilio-labs/paste#2764] [Q&A] [answered]
I'm seeing strange behavior as illustrated in this video. When a modal is opened, the width of my page container is changing. Can anyone think of anything off the top of their head that might cause this or point me in a direction to debug?

![ezgif-1-fbe0d86bc4](https://user-images.githubusercontent.com/1897006/197582960-c4c49b31-5803-46bb-82a1-819b9f848262.gif)

```jsx
import React, { useLayoutEffect, useState } from 'react';
import { useUID } from '@twilio-paste/core/uid-library';
import { Box } from '@twilio-paste/core/box';
import { Button } from '@twilio-paste/core/button';
import { Modal, ModalBody } from '@twilio-paste/core/modal';
import { SearchIcon } from '@twilio-paste/icons/esm/SearchIcon';

export default function PagefindSearchButton(): JSX.Element {
  const [isOpen, setIsOpen] = useState(false);
  const handleOpen = () => setIsOpen(true);
  const handleClose = () => setIsOpen(false);
  const searchElementID = `search${useUID()}`;
  const searchElementRef = React.useRef<HTMLDivElement>(null);

  const noop = (): void => {
    // Empty cleanup function
  };

  useLayoutEffect(() => {
    if (!isOpen) return noop;

    const timeoutId = setTimeout(() => {
      // eslint-disable-next-line no-new
      new PagefindUI({
        element: `#${searchElementID}`,
        showImages: false,
        resetStyles: false,
        showEmptyFilters: false,
        bundlePath: '/client-scripts/', // use our custom API wrapper
      });
    });

    return () => clearTimeout(timeoutId);
  }, [isOpen, searchElementID, searchElementRef]);

  return (
    <>
      <Box
        display="flex"
        alignItems="flex-end"
        columnGap="space50"
        marginLeft="space100"
      >
        <Button
          variant="secondary"
          disabled={false}
          onClick={handleOpen}
        >
          <SearchIcon
            decorative={true}
            title="Search"
          />
          Search...
        </Button>
      </Box>
      <Modal
        ariaLabelledby={searchElementID}
        isOpen={isOpen}
        onDismiss={handleClose}
        size="wide"
      >
        <ModalBody>
          <div
            id={searchElementID}
            ref={searchElementRef}
          />
        </ModalBody>
      </Modal>
    </>
  );
}
PagefindSearchButton.displayName = 'PagefindSearchButton';
```

**Comment by dprothero** [comment]
It seems to be adding padding to the `body`, which is what's causing the issue.
![image](https://user-images.githubusercontent.com/1897006/197593383-b374f735-d579-41a1-a558-d578b7196670.png)

None of which seems necessary... if I remove it, everything looks the way I would expect without modifying the layout of my page.

**Comment by dprothero** [comment]
More information: I have `max-width: 1600px;` on my `body`. If my window is resized smaller than that, I don't see the shifting.

**Reply by gloriliale** [reply]
Hi @dprothero! That's very strange; I'm talking to the team about why it could be happening. For this last detail, what is it that's setting `max-width: 1600px;` on `body`?

**Reply by gloriliale** [reply]
@dprothero Could you send the code for the page that this `PagefindSearchButton` is used in? Feel free to send it over to me directly if you'd rather not post it here. Thanks!

**Reply by dprothero** [reply]
I set the max-width myself. It's modeled after how the twilio.com marketing website is formatted.

The source for that page is not super intuitive to grok because it's mainly controlled by [Docusaurus](https://docusaurus.io/). I can only add [custom CSS](https://github.com/twilio-internal/internal-product-docs/blob/main/src/css/custom.scss) and React components. The React component for the search button is above.

**Reply by SiTaggart** [reply]
As you deploy on Netlify, maybe you can share with us the preview deployment link? Can probably debug in the browser with the compiled code to start with.

**Reply by dprothero** [reply]
@SiTaggart https://deploy-preview-355--internal-product-docs.netlify.app/ - click the Search button in the upper right.

**Reply by TheSisb** [reply]
Hi @dprothero, our Modal is based on reach/Modal which [hides the scrollbar when the modal is visible](https://github.com/reach/reach-ui/blob/dev/packages/dialog/src/reach-dialog.tsx#L266). This can sometimes cause these kinds of side effects when you have your own styles on the `body` tag.  I'd generally recommend moving some styles around to fix the issue:

- Move the `max-width: 1600px` into the child of the `body` element. In this case you can add it to `#__docusaurus` or even its child under that.
- Give that same element `margin: 0 auto`

That seems to fix the issue in my inspector. Hope this helps!

**Reply by dprothero** [reply]
That did the trick @TheSisb !

**Comment by dprothero** [comment, marked as answer]
Answer from @TheSisb:

Hi @dprothero , our Modal is based on reach/Modal which [hides the scrollbar when the modal is visible](https://github.com/reach/reach-ui/blob/dev/packages/dialog/src/reach-dialog.tsx#L266). This can sometimes cause these kinds of side effects when you have your own styles on the `body` tag. I'd generally recommend moving some styles around to fix the issue:

- Move the `max-width: 1600px` into the child of the body element. In this case you can add it to `#__docusaurus` or even its child under that.
- Give that same element `margin: 0 auto`

That seems to fix the issue in my inspector. Hope this helps!
